### DISCUSSION TITLE
New Threading Model for NonStop - Best version for PR

### DISCUSSION THREAD

**Discussion by rsbeckerca** [openssl/openssl#24072] [General Discussion]
Hi Team,
I am working on a new port for OpenSSL for a threading model on NonStop  - cannot go into details right now on it because it is not released and I'm under NDA. Testing should start soon. The changes should be entirely contained in `Configurations/50-nonstop`. What series would be most appropriate for a PR on this? I would like to include it in 3.0 if possible.

Figure roughly 3-6 months before I can complete the testing.

Opinions?
Thanks,
Randall

**Comment by nhorman** [comment]
A new threading model approach is the sort of thing that we're going to want to introduce early in a development cycle, near the beginning.6 months out from now, looking at the schedule here:
https://github.com/orgs/openssl/projects/11/views/3
places us very close to the release date for 3.4

so I would say the target for this should be 3.5, we can review/merge the changes very early in the 3.5 devel cycle and maximize testing time.

**Comment by rsbeckerca** [comment]
Thanks. When testing is done, assuming a positive outcome, I will create a PR on master and let the commits fall where them may. :)

**Comment by nhorman** [comment]
roger

**Comment by rsbeckerca** [comment]
> roger

I am not sure how long I can wait. Downstream threaded users for OpenSSL have some fairly urgent timeframes to get this in - not quantified but I might know more in 2 weeks. It may come down to having to use my fork, which is not palatable for some users. Just a heads up on this.
